packages/react/src/menu/root/MenuRoot.test.tsx

```diff
@@ -877,6 +877,46 @@ describe('<Menu.Root />', () => {
     });
   });
 
+  describe('prop: action', () => {
+    it('unmounts the menu when the `unmount` method is called', async () => {
+      const actionRef = {
+        current: {
+          unmount: spy(),
+        },
+      };
+
+      await render(
+        <Menu.Root action={actionRef}>
+          <Menu.Trigger>Open</Menu.Trigger>
+          <Menu.Portal>
+            <Menu.Positioner>
+              <Menu.Popup />
+            </Menu.Positioner>
+          </Menu.Portal>
+        </Menu.Root>,
+      );
+
+      const trigger = screen.getByRole('button', { name: 'Open' });
+      await user.click(trigger);
+
+      await waitFor(() => {
+        expect(screen.queryByRole('menu')).not.to.equal(null);
+      });
+
+      await user.click(trigger);
+
+      await waitFor(() => {
+        expect(screen.queryByRole('menu')).not.to.equal(null);
+      });
+
+      act(() => actionRef.current.unmount());
```

Comment:
In the future version of React Testing Library `act` will require awaiting. 

```suggestion
      await act(async () => actionRef.current.unmount());
```